### DISCUSSION TITLE
Spanish medical model

### DISCUSSION THREAD

**Discussion by rmasad** [deepgram/community#896] [Feature Requests]
I'm trying Deepgram as an alternative for medical voice recognition in Spanish. Other APIs (such as Watson, Google and Azure) allow you to upload sample texts so that the best word can be selected depending on the context.

I see that Deepgram can use keywords, but it is limited, both in quantity, and when testing it, I see that it replaces words that do not make sense in that context.

Is there an alternative?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by rmasad** [comment]
The `request_id` is `fc529a65-8bde-4631-8f65-9bf17cd259e6`.

Here is the code I use:

```python
import os
from dotenv import load_dotenv
import logging
from deepgram.utils import verboselogs
from datetime import datetime
import httpx

from deepgram import (
    DeepgramClient,
    DeepgramClientOptions,
    PrerecordedOptions,
    FileSource,
)

load_dotenv()

AUDIO_FILE = "audios/1.wav"

DEEPGRAM_API_KEY = os.getenv("DEEPGRAM_API_KEY")

words = [
  ...
]


keywords = [f"{word.lower()}:3" for word in words]


def main():
    try:
        # STEP 1 Create a Deepgram client using the API key in the environment variables
        config: DeepgramClientOptions = DeepgramClientOptions(
            verbose=verboselogs.SPAM,
        )
        deepgram: DeepgramClient = DeepgramClient(DEEPGRAM_API_KEY, config)
        # OR use defaults
        # deepgram: DeepgramClient = DeepgramClient()

        # STEP 2 Call the transcribe_file method on the rest class
        with open(AUDIO_FILE, "rb") as file:
            buffer_data = file.read()

        payload: FileSource = {
            "buffer": buffer_data,
        }

        options: PrerecordedOptions = PrerecordedOptions(
            model="nova-2",
            language="es-419",
            dictation=True,
            punctuate=True,
            keywords=keywords,
        )

        before = datetime.now()
        response = deepgram.listen.rest.v("1").transcribe_file(
            payload, options, timeout=httpx.Timeout(300.0, connect=10.0)
        )
        after = datetime.now()

        print(response.to_json(indent=4))
        print("")
        difference = after - before
        print(f"time: {difference.seconds}")

    except Exception as e:
        print(f"Exception: {e}")


if __name__ == "__main__":
    main()
```

**Comment by jpvajda** [comment]
@rmasad  if you are interested in custom model training for advanced key word use cases that is something available to our Enterprise Customers only at this time. I can put you in touch with someone to discuss that option with you, just let me know. 

I'm also curious if a Spanish medical model be what you are actually in need of instead of custom model training.

**Comment by rmasad** [comment]
I checked your documentation and it seems that the medical model is not available in Spanish for testing.

**Reply by jpvajda** [reply]
@rmasad  We don't offer a medical model yet for Spanish, but I was asking if what you are really looking for is just a Spanish medical model instead of trying to do custom training of spanish model with your own list of medical related keywords.

I don't have an ETA of when we'll have a Spanish medical model, but I can mention your interest to our product team.

**Reply by rmasad** [reply]
It could be interesting, depending on the WER that results in the dataset we have.

**Comment by rmasad** [comment]
Coming back to this. Is it part of your plans to include medical model in Spanish?
